frontend/src/pages/Status.test.tsx

```diff
@@ -48,6 +48,256 @@ describe('Status', () => {
       expect(consoleSpy).toHaveBeenLastCalledWith('Unknown node phase:', undefined);
     });
 
+    // TODO: Enable this test after react-scripts is upgraded to v4.0.0
+    // it('react testing for ERROR phase', async () => {
+    //   const { findByText, getByTestId } = render(
+    //     statusToIcon(NodePhase.ERROR),
+    //   );
+
+    //   fireEvent.mouseOver(getByTestId('node-status-sign'));
+    //   findByText('Error while running this resource');
+    // });
+
+    it('handles ERROR phase', () => {
+      const tree = shallow(statusToIcon(NodePhase.ERROR));
+      expect(tree.find('span')).toMatchInlineSnapshot(`
+        <span
+          style={
+            Object {
+              "height": 18,
+            }
+          }
+        >
+          <pure(ErrorIcon)
+            data-testid="node-status-sign"
+            style={
+              Object {
+                "color": "#d50000",
+                "height": 18,
+                "width": 18,
+              }
+            }
+          />
+        </span>
+      `);
+    });
+
+    it('handles FAILED phase', () => {
+      const tree = shallow(statusToIcon(NodePhase.FAILED));
+      expect(tree.find('span')).toMatchInlineSnapshot(`
+        <span
+          style={
+            Object {
+              "height": 18,
+            }
+          }
+        >
+          <pure(ErrorIcon)
+            data-testid="node-status-sign"
+            style={
+              Object {
+                "color": "#d50000",
+                "height": 18,
+                "width": 18,
+              }
+            }
+          />
+        </span>
+      `);
+    });
+
+    it('handles PENDING phase', () => {
+      const tree = shallow(statusToIcon(NodePhase.PENDING));
+      expect(tree.find('span')).toMatchInlineSnapshot(`
```

Comment:
this is an unstable finder, if any other `span` element shows up, it'll break.

I'd suggest this blogpost https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change.
There are some general advice for choosing a selector, it should be
1. reflect how users find elements
2. if not feasible, add a unique data-testid

This is not a blocker for this PR, your improvements have been pretty good!

Reply:
Thank you for the suggestion! I agree with the testing approach which is based on user visible element. The current implementation is temporary, and we should use the react testing library (commented test) and abandon this one.